lib/client.js

```diff
@@ -180,35 +181,37 @@ api._pollServer = function(options, callback) {
 };
 
 function pollServer(options, callback) {
-  console.log("Poll server called with options", options);
   var client = options.client;
-  var retries = 0;
-  var maxRetries = config['messages-client'].maxMessageRetrievalRetries;
-  var failedToRetrieveMessageEvent = {
-    type: 'messageClient.failedToRetrieveMessage',
-    details: {
-      endpoint: client.endpoint
-    }
-  };
+  var jobId = createJobId(client);
+  var retries;
+
   async.auto({
-    getPrivateKey: function(callback) {
+    getClient: function(callback) {
+      database.collections.messageClient.findOne(
+        {id: database.hash(client.id)}, {}, function(err, record) {
+          retries = record.meta.retries;
+          callback(null, record);
+        });
+    },
+    getPrivateKey: function(callback, result) {
       brKey.getPublicKey({id: client.publicKeyId}, null,
         function(err, publicKey, meta, privateKey) {
-          if(!privateKey) {
-            return callback(new BedrockError(
-            'Contacting server with no private key configured: ' +
-              client.endpoint,
-            'SigningFailure', {statusCode: 400}
-            ));
+          if(!privateKey || err) {
+            return handlePollServerError({
+              client: client,
+              jobId: jobId,
+              error: new BedrockError(
+                'The server could not be contacted: ' + client.endpoint,
+                'HttpError', {
```

Comment:
This doesn't follow any conventions we have elsewhere. It should be using `httpStatusCode` in the details and using an appropriate error name (see if you can find some other examples) that is a little more semantic with regard to the action you were trying to take.